runtime/jcl/common/thread.cpp

```diff
@@ -280,15 +280,25 @@ Java_java_lang_Thread_interruptImpl(JNIEnv *env, jobject rcv)
 	vmFuncs->internalEnterVMFromJNI(currentThread);
 	j9object_t receiverObject = J9_JNI_UNWRAP_REFERENCE(rcv);
 	J9VMThread *targetThread = J9VMJAVALANGTHREAD_THREADREF(currentThread, receiverObject);
-	if (J9VMJAVALANGTHREAD_STARTED(currentThread, receiverObject)) {
-		if (NULL != targetThread) {
+	if (NULL != targetThread) {
+		if (J9VMJAVALANGTHREAD_STARTED(currentThread, receiverObject)) {
 			void (*sidecarInterruptFunction)(J9VMThread*) = vm->sidecarInterruptFunction;
 			if (NULL != sidecarInterruptFunction) {
 				sidecarInterruptFunction(targetThread);
 			}
 			omrthread_interrupt(targetThread->osThread);
 		}
+	} else if (J2SE_VERSION(vm) >= J2SE_V14) {
+		static jclass globalClassRef = (*env)->NewGlobalRef(env, (*env)->GetObjectClass(env, rcv));
+		static jfieldID fid = (*env)->GetFieldID(env, globalClassRef, "deadInterrupt", "Z");
```

Comment:
We don't use static variables for this, there is a structure, however you can use the macro J9VMJAVALANGTHREAD_DEADINTERRUPT rather than using JNI, so the global ref/field id isn't required anyway.

Reply:
J9VMJAVALANGTHREAD_SET_DEADINTERRUPT I meant.

Reply:
Agreed and replaced with `J9VMJAVALANGTHREAD_SET_DEADINTERRUPT`.